### PR TITLE
Opening cleanups

### PR DESCRIPTION
**Pull request by rustyrussell**
Another major series, this one on top of #975 

The only remaining cleanup I didn't do, was to split peer_control.c.  That's probably worth a standalone patch; this one is already in danger of needing constant rebasing!

### CODE REVIEW

File: channeld/channel.c [hunk 1]
```diff
@@ -1557,8 +1564,8 @@ static void peer_in(struct peer *peer, const u8 *msg)
 	/* Must get funding_locked before almost anything. */
 	if (!peer->funding_locked[REMOTE]) {
 		if (type != WIRE_FUNDING_LOCKED && type != WIRE_PONG) {
-			peer_failed(PEER_FD,
-				    &peer->cs,
+			peer_failed(	    &peer->cs,
```
Comment (ZmnSCPxj) [inline channeld/channel.c]: extra whitespace

Comment (rustyrussell) [reply channeld/channel.c]: Thanks, fixed!

File: lightningd/peer_control.c [hunk 2]
```diff
@@ -278,6 +362,63 @@ static void channel_config(struct lightningd *ld,
...
+	 *  - when sending `error`:
+	 *    - MUST fail the channel referred to by the error message.
+	 *...
+	 * The receiving node:
+	 *  - upon receiving `error`:
+	 *    - MUST fail the channel referred to by the error message.
+	 */
+	channel_fail_permanent(channel, "%s: %s ERROR %s",
+			       channel->owner->name,
+			       err_for_them ? "sent" : "received", desc);
```
Comment (ZmnSCPxj) [inline lightningd/peer_control.c]: Is this `desc` the raw data from the `error` message? How do we know, this is safely printable to our log? Or, already filtered at this point?

**Comment by cdecker** [comment]
Sorry for the delay reviewing this, but I tried to rebase on top of `master` to prune commits that are already in there. Now `valgrind` complains again :frowning_face:

File: lightningd/peer_control.c (hunk 2 above)
Comment (rustyrussell) [reply lightningd/peer_control.c]: Indeed, already filtered.  desc comes from sanitize_error(), which does this work for us.  It also comes through wirestring, marshalling, which similarly fails on weird stuff.

**Comment by rustyrussell** [comment]
OK, @cdecker I'm taking your rebase as approval, and am testing rebase now.   If all good, will merge.

File: lightningd/peer_control.c (hunk 2 above)
Comment (rustyrussell) [reply lightningd/peer_control.c]: BTW, @ZmnSCPxj I sent mail to your protonmail, not sure how often you check it though...

Comment (ZmnSCPxj) [reply lightningd/peer_control.c]: Not very often, but have seen and replied. Thank you very much your email.

**Comment by ZmnSCPxj** [comment]
Some curiousity, but what is `tal_link`? Am I right in thinking, it is something similar to `std::shared_ptr`, i.e. multiple sharing owners with effectively something similar to refcount (except using a linked list instead of counter)?  How does it interact with `tal_steal`?

**Pull request merged**